### PR TITLE
pythonPackages.customtkinter: init at 5.2.2

### LINKED ISSUE #235337: Package request: CustomTkinter
**Project description**
A modern and customizable python UI-library based on Tkinter

**Metadata**

* homepage URL: https://customtkinter.tomschimansky.com/
* source URL: https://github.com/TomSchimansky/CustomTkinter
* license: mit,
* platforms: unix, linux, darwin, ...

### PR DESCRIPTION
**Pull request by 4evy**
## Description of changes

Closes #235337

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 306344` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.customtkinter</li>
    <li>python311Packages.customtkinter.dist</li>
    <li>python312Packages.customtkinter</li>
    <li>python312Packages.customtkinter.dist</li>
  </ul>
</details>

**Comment by wegank** [comment]
@ofborg build python311Packages.customtkinter python311Packages.customtkinter.passthru.tests

**Comment by DontEatOreo** [comment]
All checks passed @wegank

**Review by wegank — APPROVED** [review]

File: pkgs/development/python-modules/customtkinter/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  buildInputs = [ tkinter ];
+  propagatedBuildInputs = [ darkdetect ];
+
+  doCheck = false;
+  pythonImportsCheck = [ "customtkinter" ];
+
+  meta = {
+    description = "A modern and customizable python UI-library based on Tkinter";
+    longDescription = ''
+      CustomTkinter is a python UI-library based on Tkinter, which provides new, modern and fully customizable widgets. They are created and used like normal Tkinter widgets and can also be used in combination with normal Tkinter elements. The widgets and the window colors either adapt to the system appearance or the manually set mode ('light', 'dark'), and all CustomTkinter widgets and windows support HighDPI scaling (Windows, macOS). With CustomTkinter you'll get a consistent and modern look across all desktop platforms (Windows, macOS, Linux).
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/customtkinter/default.nix:32]:
```suggestion
      CustomTkinter is a python UI-library based on Tkinter, which provides
      new, modern and fully customizable widgets. They are created and
      used like normal Tkinter widgets and can also be used in combination
      with normal Tkinter elements. The widgets and the window colors
      either adapt to the system appearance or the manually set mode
      ('light', 'dark'), and all CustomTkinter widgets and windows support
      HighDPI scaling (Windows, macOS). With CustomTkinter you'll get
      a consistent and modern look across all desktop platforms
      (Windows, macOS, Linux).
```

This looks more comfortable. It does not affect the arrangement on nixos-search, as a newline will be converted into a space.

File: pkgs/development/python-modules/customtkinter/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-/Y2zuvqWHJgu5gMNuoC0wuJYWGMHVrUTmG2xkRPY0gc=";
+  };
+
+  nativeBuildInputs = [ setuptools wheel ];
+  buildInputs = [ tkinter ];
+  propagatedBuildInputs = [ darkdetect ];
+
+  doCheck = false;
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/customtkinter/default.nix:26]: Please add an explanation in the comment above.

File: pkgs/development/python-modules/customtkinter/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+, setuptools
+, wheel
+, tkinter
+, darkdetect
+}:
+
+buildPythonPackage rec {
+  pname = "customtkinter";
+  version = "5.2.2";
+  format = "pyproject";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/customtkinter/default.nix:14]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/customtkinter/default.nix (hunk 1 above)
Comment (DontEatOreo) [reply pkgs/development/python-modules/customtkinter/default.nix]: Done

File: pkgs/development/python-modules/customtkinter/default.nix (hunk 2 above)
Comment (DontEatOreo) [reply pkgs/development/python-modules/customtkinter/default.nix:31]: Done

File: pkgs/development/python-modules/customtkinter/default.nix (hunk 3 above)
Comment (DontEatOreo) [reply pkgs/development/python-modules/customtkinter/default.nix]: Done

**Comment by DontEatOreo** [comment]
I also ran `nixfmt-rfc-style` on the package to comply with RFC 0166

File: pkgs/development/python-modules/customtkinter/default.nix [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-/Y2zuvqWHJgu5gMNuoC0wuJYWGMHVrUTmG2xkRPY0gc=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/customtkinter/default.nix:27]: We don't need wheel here, do we?

File: pkgs/development/python-modules/customtkinter/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/customtkinter/default.nix:33]: that's just not true https://github.com/TomSchimansky/CustomTkinter/tree/master/test

They are likely just not uploaded to pypi and we need to fetch from github to get them

Comment (DontEatOreo) [reply pkgs/development/python-modules/customtkinter/default.nix:33]: Okay, to clarify, I meant in the context of the `fetchPypi` as there are no tests that come from the PyPI.

If necessary, I could try packaging it from the source and include the tests?

**Pull request merged**